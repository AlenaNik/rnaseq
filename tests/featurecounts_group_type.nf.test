nextflow_pipeline {

    name "Test pipeline without a group type for featureCounts"
    script "../main.nf"

    test("Params: --featurecounts_group_type false") {

        when {
            params {
                outdir = "$outputDir"
                featurecounts_group_type = false
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${params.outdir}/bbsplit/RAP1_IAA_30M_REP1.stats.txt"),
                    path("${params.outdir}/bbsplit/RAP1_UNINDUCED_REP1.stats.txt"),
                    path("${params.outdir}/bbsplit/RAP1_UNINDUCED_REP2.stats.txt"),
                    path("${params.outdir}/bbsplit/WT_REP1.stats.txt"),
                    path("${params.outdir}/bbsplit/WT_REP2.stats.txt")
                ).match("bbsplit") },
                { assert snapshot(
                    path("${params.outdir}/custom/out/genome_gfp.fasta"),
                    path("${params.outdir}/custom/out/genome_gfp.gtf")
                ).match("references") },
                { assert snapshot(
                    path("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw_1_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw_2_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP1_raw_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP2_raw_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/WT_REP1_raw_1_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/WT_REP1_raw_2_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/WT_REP2_raw_1_fastqc.html"),
                    path("${params.outdir}/fastqc/raw/WT_REP2_raw_2_fastqc.html"),
                    // ZIPs are not stable
                    file("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw_2_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP1_raw_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP2_raw_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/WT_REP1_raw_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/WT_REP1_raw_2_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/WT_REP2_raw_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/raw/WT_REP2_raw_2_fastqc.zip").name
                ).match("fastqc/raw") },
                { assert snapshot(
                    path("${params.outdir}/fastqc/trim/RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/WT_REP1_trimmed_1_val_1_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/WT_REP1_trimmed_2_val_2_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/WT_REP2_trimmed_1_val_1_fastqc.html"),
                    path("${params.outdir}/fastqc/trim/WT_REP2_trimmed_2_val_2_fastqc.html"),
                    // ZIPs are not stable
                    file("${params.outdir}/fastqc/trim/RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/WT_REP1_trimmed_1_val_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/WT_REP1_trimmed_2_val_2_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/WT_REP2_trimmed_1_val_1_fastqc.zip").name,
                    file("${params.outdir}/fastqc/trim/WT_REP2_trimmed_2_val_2_fastqc.zip").name
                ).match("fastqc/trim") }
            )
        }
    }
}
