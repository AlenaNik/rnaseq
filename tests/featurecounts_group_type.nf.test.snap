{
    "references": {
        "content": [
            "genome_gfp.fasta:md5,e23e302af63736a199985a169fdac055",
            "genome_gfp.gtf:md5,b4776accd1bf7206c7ab0f1d84e4721e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T15:44:10.068551"
    },
    "fastqc/trim": {
        "content": [
            "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.html:md5,3823cf549c84bd16c0a5aa42fa4853e7",
            "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.html:md5,874a7e78c3ec55eec7b7b70b661a27f2",
            "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.html:md5,c9a605d6c5132c0b37ea22437a94d21e",
            "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.html:md5,47510573cc288ff174be16704ae8fce8",
            "WT_REP1_trimmed_1_val_1_fastqc.html:md5,dd7deefb9d924cd452c0e8a1789907cc",
            "WT_REP1_trimmed_2_val_2_fastqc.html:md5,efbadd5f6f79a2be7ee1ca9cdf12c3a2",
            "WT_REP2_trimmed_1_val_1_fastqc.html:md5,113b4b6e3aea6d0cbe533a4f67b47b75",
            "WT_REP2_trimmed_2_val_2_fastqc.html:md5,10407149a3a88b5be77fe3b2a090e4c6",
            "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.zip",
            "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.zip",
            "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.zip",
            "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.zip",
            "WT_REP1_trimmed_1_val_1_fastqc.zip",
            "WT_REP1_trimmed_2_val_2_fastqc.zip",
            "WT_REP2_trimmed_1_val_1_fastqc.zip",
            "WT_REP2_trimmed_2_val_2_fastqc.zip"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T15:44:10.142955"
    },
    "bbsplit": {
        "content": [
            "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
            "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
            "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
            "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de",
            "WT_REP2.stats.txt:md5,002ddf6b828d34979652a40f05462843"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T15:44:10.05569"
    },
    "fastqc/raw": {
        "content": [
            "RAP1_IAA_30M_REP1_raw_1_fastqc.html:md5,4d0fd346e2d8b8083ba1d7b70179f735",
            "RAP1_IAA_30M_REP1_raw_2_fastqc.html:md5,b4b3096f031a7d7b51cf45800c9df893",
            "RAP1_UNINDUCED_REP1_raw_fastqc.html:md5,fc4bf32d182aaf26c3d5d66a2e1e72d9",
            "RAP1_UNINDUCED_REP2_raw_fastqc.html:md5,cf53cc7f31f21541212382f832dedd8d",
            "WT_REP1_raw_1_fastqc.html:md5,d78401014c733eea1e560af7a60838ed",
            "WT_REP1_raw_2_fastqc.html:md5,21f06f45712b298fda165a8a3cf3acde",
            "WT_REP2_raw_1_fastqc.html:md5,c41d4f91d622edc9a9027f1c0cef2b63",
            "WT_REP2_raw_2_fastqc.html:md5,c9ce8891cb65d72c6cd2df960f02b81c",
            "RAP1_IAA_30M_REP1_raw_1_fastqc.zip",
            "RAP1_IAA_30M_REP1_raw_2_fastqc.zip",
            "RAP1_UNINDUCED_REP1_raw_fastqc.zip",
            "RAP1_UNINDUCED_REP2_raw_fastqc.zip",
            "WT_REP1_raw_1_fastqc.zip",
            "WT_REP1_raw_2_fastqc.zip",
            "WT_REP2_raw_1_fastqc.zip",
            "WT_REP2_raw_2_fastqc.zip"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-22T15:44:10.08366"
    }
}